OJS.Workers.ExecutionStrategies/NodeJs/NodeJsProjectRunJavaScriptProjectAndTestsWithPlaywrigth.cs

```diff
@@ -0,0 +1,279 @@
+                result.CompilerComment = exception.Message;
+
+                return result;
+            }
+
+            this.SaveTestsToFiles(executionContext.Input.Tests);
+            this.SaveNginxFile();
+
+            var codeSavePath = this.SavePythonCodeTemplateToTempFile();
+            var executor = this.CreateExecutor();
+            var checker = executionContext.Input.GetChecker();
+            return this.RunTests(codeSavePath, executor, checker, executionContext, result);
+        }
+
+        protected override IExecutionResult<TestResult> RunTests(
+            string codeSavePath,
+            IExecutor executor,
+            IChecker checker,
+            IExecutionContext<TestsInputModel> executionContext,
+            IExecutionResult<TestResult> result)
+        {
+            foreach (var test in executionContext.Input.Tests)
+            {
+                var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, test.Input);
+                var testResults = this.ExtractTestResultsFromReceivedOutput(processExecutionResult, test.Id);
+                result.Results.AddRange(testResults);
+            }
+
+            return result;
+        }
+
+        protected override IExecutor CreateExecutor() => this.CreateExecutor(ProcessExecutorType.Standard);
+
+        private void ExtractSubmissionFiles<TInput>(IExecutionContext<TInput> executionContext)
+        {
+            
+            var trimmedAllowedFileExtensions = executionContext.AllowedFileExtensions?.Trim();
+            var allowedFileExtensions = (!trimmedAllowedFileExtensions?.StartsWith(".") ?? false)
+                ? $".{trimmedAllowedFileExtensions}"
+                : trimmedAllowedFileExtensions;
+
+            if (allowedFileExtensions != Constants.ZipFileExtension)
+            {
+                throw new ArgumentException("Submission file is not a zip file!");
+            }
+
+            var submissionFilePath = FileHelpers.BuildPath(this.WorkingDirectory, "temp");
+            File.WriteAllBytes(submissionFilePath, executionContext.FileContent);
+            FileHelpers.RemoveFilesFromZip(submissionFilePath, RemoveMacFolderPattern);
+            FileHelpers.UnzipFile(submissionFilePath, this.UserApplicationPath);
+
+            System.IO.Directory.CreateDirectory(this.UserApplicationPath);
+            System.IO.Directory.CreateDirectory(this.TestsPath);
+        }
+
+        private ICollection<TestResult> ExtractTestResultsFromReceivedOutput(ProcessExecutionResult executionResult, int parentTestId)
+        {
+            // Unescape mocha output
+            var receivedOutput = Regex.Unescape(executionResult.ReceivedOutput);
+            receivedOutput = receivedOutput.Replace("b'", string.Empty);
+            receivedOutput = receivedOutput.Replace("}'", "}");
+
+            dynamic deserializedOutput = JsonConvert.DeserializeObject(receivedOutput);
+            var testResults = deserializedOutput.tests;
```

Comment:
I would prefer to extract this into a method, along with line 224.

This will keep all `json`/`dynamic` code isolated into a single method and this method will work with POCO